Formula/raxml-ng.rb

```diff
@@ -43,21 +43,30 @@ class RaxmlNg < Formula
 
   def install
     args = std_cmake_args + ["-DUSE_GMP=ON"]
-    mkdir "build" do
-      system "cmake", "..", *args
-      system "make", "install"
-    end
-    mkdir "build_mpi" do
-      ENV["CC"] = "mpicc"
-      ENV["CXX"] = "mpicxx"
-      system "cmake", "..", *args, "-DUSE_MPI=ON", "-DRAXML_BINARY_NAME=raxml-ng-mpi"
-      system "make", "install"
-    end
+    system "cmake", "-S", ".", "-B", "build", *args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    # Upstream doesn't support building MPI variant on macOS.
+    # The build ignores USE_MPI=ON and forces ENABLE_MPI=OFF.
+    # This causes necessary flags like -D_RAXML_MPI to not get set.
+    return if OS.mac?
+
+    system "cmake", "-S", ".", "-B", "build_mpi", *args, "-DUSE_MPI=ON"
+    system "cmake", "--build", "build_mpi"
+    system "cmake", "--install", "build_mpi"
   end
 
   test do
     testpath.install resource("homebrew-example")
-    system "#{bin}/raxml-ng", "--msa", "dna.phy", "--start", "--model", "GTR"
+    system bin/"raxml-ng", "--msa", "dna.phy", "--start", "--model", "GTR"
+    return if OS.mac?
+
+    mkdir "test-mpi" do
+      Pathname.pwd.install resource("homebrew-example")
+      assert_match "parallelization: coarse-grained (auto), hybrid MPI (2 ranks) + PTHREADS (auto)",
+                   shell_output("mpirun -np 2 #{bin}/raxml-ng-mpi --msa dna.phy --start --model GTR")
```

Comment:
Also hit an error with existing bottle so need to find out if I am running MPI variant incorrectly.